File: Java/programmers/해시/위장.java
package programmers.해시;

import java.util.HashMap;

public class 위장 {
    public int solution(String[][] clothes) {
        HashMap<String, Integer> hm = new HashMap<>();
        for(int i = 0; i <clothes.length; i++){
            if(!hm.containsKey(clothes[i][1])) hm.put(clothes[i][1],1);
            else hm.put(clothes[i][1] , hm.get(clothes[i][1])+1);
        }
        int answer = 1;
        for(String s : hm.keySet()){
            answer*=hm.get(s)+1;
        }
        return answer-1;
    }
}

File: JavaScript/programmers/위장.js
function solution(clothes) {
    var answer = 1;
    let map = {};
    for(let i = 0; i < clothes.length;  i++){
        if(map[clothes[i][1]]){
            map[clothes[i][1]]++;
        }else{
            map[clothes[i][1]] = 1;
        }
    }
    
    for(const el in map) answer*= map[el] + 1;
    
    return answer - 1;
}